### PR TITLE
Webkit Changes for SDK Core

### PR DESCRIPTION
**Pull request by trooper2013**
Changes for the replacement of UIWebView with WKWebView.

### CODE REVIEW

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.h [hunk 1]
```diff
@@ -24,12 +24,16 @@
...
+#import <WebKit/WebKit.h>
 #import "SFOAuthCredentials.h"
 
 
 @class SFOAuthCoordinator;
 @class SFOAuthInfo;
 
+
+
+
```
Comment (kchitalia) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.h]: Do we really need these?

File: libs/SalesforceHybridSDK/SalesforceHybridSDK.xcodeproj/xcshareddata/xcschemes/SalesforceHybridSDKTestAppTests.xcscheme [hunk 2]
```diff
@@ -5,6 +5,19 @@
    <BuildAction
       parallelizeBuildables = "YES"
       buildImplicitDependencies = "YES">
+      <BuildActionEntries>
```
Comment (bhariharan) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK.xcodeproj/xcshareddata/xcschemes/SalesforceHybridSDKTestAppTests.xcscheme]: I think this can be reverted.

File: libs/SalesforceNetwork/SalesforceNetwork.xcodeproj/xcshareddata/xcschemes/SalesforceNetworkTests.xcscheme [hunk 3]
```diff
@@ -5,6 +5,19 @@
    <BuildAction
       parallelizeBuildables = "YES"
       buildImplicitDependencies = "YES">
+      <BuildActionEntries>
```
Comment (bhariharan) [inline libs/SalesforceNetwork/SalesforceNetwork.xcodeproj/xcshareddata/xcschemes/SalesforceNetworkTests.xcscheme]: I think this can be reverted.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Login/SFLoginViewController.m [hunk 4]
```diff
@@ -175,7 +176,7 @@ - (SFSDKLoginHostListViewController *)loginHostListViewController {
 
 #pragma mark - Properties
 
-- (void)setOauthView:(UIView *)oauthView {
+- (void)setOauthView:(WKWebView *)oauthView {
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Login/SFLoginViewController.m]: I don't think we should change this API. Let's leave it as ```UIView```.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Login/SFLoginViewController.m [hunk 5]
```diff
@@ -27,13 +27,14 @@
  */
 
 @import UIKit;
-
+#import <WebKit/WebKit.h>
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Login/SFLoginViewController.m]: This import can go away.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m [hunk 6]
```diff
@@ -162,7 +163,8 @@ - (void)dealloc {
     _responseData = nil;
     _scopes = nil;
     [self stopRefreshFlowConnectionTimer];
-    _view.delegate = nil;
+    [_view setNavigationDelegate:nil];
+    [_view setUIDelegate:nil];
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m]: We don't need this, right? Do we ever set ```WKUIDelegate```?

File: libs/SalesforceHybridSDK/SalesforceHybridSDK.xcodeproj/xcshareddata/xcschemes/SalesforceHybridSDKTestAppTests.xcscheme [hunk 7]
```diff
@@ -5,6 +5,19 @@
...
+            buildForTesting = "YES">
+            <BuildableReference
+               BuildableIdentifier = "primary"
+               BlueprintIdentifier = "8222729C19C39F4500FC537E"
+               BuildableName = "SalesforceHybridSDKTests.xctest"
+               BlueprintName = "SalesforceHybridSDKTests"
+               ReferencedContainer = "container:SalesforceHybridSDK.xcodeproj">
+            </BuildableReference>
+         </BuildActionEntry>
+      </BuildActionEntries>
```
Comment (kchitalia) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK.xcodeproj/xcshareddata/xcschemes/SalesforceHybridSDKTestAppTests.xcscheme]: I think this whole section can be reverted

File: libs/SalesforceNetwork/SalesforceNetwork.xcodeproj/xcshareddata/xcschemes/SalesforceNetworkTests.xcscheme [hunk 8]
```diff
@@ -5,6 +5,19 @@
...
+            buildForTesting = "YES">
+            <BuildableReference
+               BuildableIdentifier = "primary"
+               BlueprintIdentifier = "825347F41A8ADD050019FE0B"
+               BuildableName = "SalesforceNetworkTests.xctest"
+               BlueprintName = "SalesforceNetworkTests"
+               ReferencedContainer = "container:SalesforceNetwork.xcodeproj">
+            </BuildableReference>
+         </BuildActionEntry>
+      </BuildActionEntries>
```
Comment (kchitalia) [inline libs/SalesforceNetwork/SalesforceNetwork.xcodeproj/xcshareddata/xcschemes/SalesforceNetworkTests.xcscheme]: Same here

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m [hunk 9]
```diff
@@ -988,7 +989,22 @@ - (void)webViewDidFinishLoad:(UIWebView *)webView {
     }
 }
 
-- (void)webView:(UIWebView *)webView didFailLoadWithError:(NSError *)error {
+- (void)webView:(WKWebView *)webView didCommitNavigation:(WKNavigation *)navigation
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m]: Let's lose this method, since we don't do anything here. Or call the super.

Comment (kchitalia) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m]: I think its safe to remove this method as its a no op.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m (hunk 6 above)
Comment (trooper2013) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m]: We can get rid of this line sure.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFSDKWebUtils.h [hunk 10]
```diff
@@ -46,4 +46,5 @@ extern NSString * const kUserAgentPropKey;
  */
 + (NSString *)currentUserAgentForApp;
 
+
```
Comment (kchitalia) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFSDKWebUtils.h]: Please revert this.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFSDKWebUtils.m [hunk 11]
```diff
@@ -22,10 +22,10 @@
  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
  WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
-
 #import "SFSDKWebUtils.h"
 #import "SFApplication.h"
-
+#import <WebKit/WebKit.h>
+//#import "WKWebView+SFWKWebView.h"
```
Comment (kchitalia) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFSDKWebUtils.m]: Please remove this commented out line.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAuthenticationViewHandler.m [hunk 12]
```diff
@@ -23,6 +23,7 @@
  WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+#import <WebKit/WebKit.h>
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAuthenticationViewHandler.m]: I think this import can go away?

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFSDKWebUtils.h (hunk 10 above)
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFSDKWebUtils.h]: Revert changes here. Nothing has changed except a newline.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFSDKWebUtils.m (hunk 11 above)
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFSDKWebUtils.m]: Remove commented out line.

File: libs/SalesforceSDKCore/SalesforceSDKCore/SalesforceSDKCore.h [hunk 13]
```diff
@@ -2,7 +2,7 @@
  SalesforceSDKCore.h
  SalesforceSDKCore
 
- Created by Michael Nachbaur on Wed May 11 11:34:05 PDT 2016.
+ Created by Raj Rao on Mon Jul 11 17:04:07 PDT 2016.
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/SalesforceSDKCore.h]: Revert this change. Nothing's changed in this file.

**Comment by bhariharan** [comment]
Nice! 👍
